### DISCUSSION TITLE
graphic bug opening application menu

### DISCUSSION THREAD

**Discussion by sette15** [lxqt/lxqt#2105] [Q&A]
hey eveybody. 
I'm running LXQt 0.17.0 in openSUSE Tumbleweed in a really low end laptop with a Intel solo processor 32bit.
It does run quite good, however so far I do have this not major bug, but definitely annoying: everytime I click on the application menu, there is a huge "pixel gap" between the bar and the actual application menu.
I did try to change the window manager to icewm or even xfwm4, the bug still remains. 
Please see video attached showing the bug.

https://user-images.githubusercontent.com/87567555/139340719-9a15cf89-5abe-4561-8af9-876b8107263e.MOV

I'd really appreciate any suggestion, thanks!

**Comment by tsujan** [comment]
Your menu seems to have translucency. So, you should have a compositor running. Which compositor is it, other than that of xfwm4?

Does it also happen when you kill/stop the compositor?

**Reply by sette15** [reply]
Hi tsujan, thanks for your quick response.
At the moment I'm using openbox as it was by default. I tried xfwm4 just for a few minutes and now is uninstalled.
Sorry I'm quite new to LXQt and not an expert on Linux. Can you please show me how to kill/stop the compositor?
I'd also like to point out that if I try to perform any search on the application menu and delete the words, the bug disappears for the moment. Please see video attached. Thanks

https://user-images.githubusercontent.com/87567555/139532614-34ce6cf9-737b-4b63-8f27-4a49a132f804.MOV

**Reply by tsujan** [reply]
> Sorry I'm quite new to LXQt and not an expert on Linux. 

No need to be sorry. Only the basic concepts are needed and you'll get familiar with them quickly.

There are two things at work here: window manager (WM) and compositor. WM is responsible for handling all kinds of windows (menus are included too). The compositor is the program that makes translucency, shadow and, probably, some other properties (like blurring and 3D effects) possible. It also affects windows.

Some WMs, like Openbox, don't have a compositor, so that you need a separate compositor like Compton with them. Some others, like xfwm4 and KWin, come with their own compositor. LXQt doesn't provide a WM or compositor; you could use any with it.

When you use Openbox, check whether a process with the name "compton" is running. Use a process manager (like Qps or any process manager of your choice) for that. If it is running, terminate it and see if the problem persists. Please note that if you terminate it, your panel and menus should lose translucency; that's a sign.

**Comment by sette15** [comment]
thank you so much.
so, looking on Qps it doesn’t seems there is any “Compton” process, in fact I tried to install it and it gives me the opportunity to do so. I guess openSUSE doesn’t have this compositor as default.

**Reply by tsujan** [reply]
So, there should be another compositor with Openbox. The point is that you disable compositing temporarily to find whether it's the cause.

**Comment by stefonarch** [comment]
Check also for picom in qps, which is a fork of compton (which is abbandoned).

**Comment by sette15** [comment]
Hi, thank you both.
Unfortunately can't find which compositing I'm using. There is no sign of Picom either.
Is there a terminal command to define which one I'm using?

**Reply by tsujan** [reply]
Take a look at LXQt Session Settings → Autostart. You might find something there.

Please note that, by seeing your screen records, I assumed that your main menu has translucency *with Openbox*. If it is *not* translucent with Openbox but only with xfwm4, you use xfwm4's internal compositor when you switch to xfwm4, which may be disabled in its config. If it *is* translucent with Openbox, you should have a standalone compositor, which may interfere with that of xfwm4 too (only one compositor should be running).

**Comment by stefonarch** [comment]
You can also verify if there is a compositor in use by panel options > transparency, this is only active when one is running.

**Comment by sette15** [comment]
There we go. 
Apparently I’ve no compositor running as showed in the picture:


![](https://i.imgur.com/UpqWtOk.jpg)

**Reply by tsujan** [reply]
> Apparently I’ve no compositor running as showed in the picture:

It doesn't show the lack or existence of a compositor.

It's as simple as this: does your main menu have translucency?

**Reply by sette15** [reply]
I though the message "Compositing is required for panel transparency" shows there is no compositor.

Anyway, I'm not quite sure, but I think the main menu have translucency. Not 100% sure tho

**Reply by tsujan** [reply]
It should look like a glass: put any window where it appears and check if you could see the window through it partially.

**Comment by sette15** [comment]
definitely doesn’t look like a glass. it’s pretty “flat”

**Reply by tsujan** [reply]
> definitely doesn’t look like a glass. it’s pretty “flat”

If by "flat" you mean "opaque" *and* if the LXQt theme you use has translucent colors, we could be sure that you don't have a compositor running.

There should be a hidden factor somewhere. That issue has never happened here, with 5 totally different computers (all 64-bit), with or without composting, and with 4 different WMs (KWin, Openbox, compiz-reloaded and xfwm4). More importantly, I found nothing in LXQt's code that could have caused it.

It may be related to some 32-bit packages of OpenSUSE (not necessarily its LXQt packages) but that's just a wild guess.

We might need to wait for a similar report and more info, or for someone with new ideas...

**Reply by sette15** [reply]
in the meantime, thank you for your help
